ingestion/src/metadata/ingestion/source/pipeline/airflow/connection.py

```diff
@@ -114,8 +128,39 @@ def test_connection(
     of a metadata workflow or during an Automation Workflow
     """
 
-    test_fn = {"CheckAccess": partial(test_connection_engine_step, engine)}
-
+    session_maker = sessionmaker(bind=engine)
+    session = session_maker()
+
+    def test_pipeline_details_access(session):
+        try:
+            result = session.query(SerializedDagModel).limit(1)
+            if result:
+                return result
+            raise AirflowPipelineNotFound("No Pipeline found")
+        except Exception as e:
+            raise AirflowPipelineNotFound(f"Pipeline details access error : {e}")
+
+    def test_task_detail_access(session):
+        try:
+            json_data_column = (
+                SerializedDagModel._data  # For 2.3.0 onwards # pylint: disable=protected-access
+                if hasattr(SerializedDagModel, "_data")
+                else SerializedDagModel.data  # For 2.2.5 and 2.1.4
+            )
+            result = session.query(json_data_column).limit(1).all()
```

Comment:
instead of using limit and all, we could use first
wdyt?

Reply:
ok let me try.

Reply:
Did the required changes, tested using CLI and UI, works fine.